### PR TITLE
Add demo of tracking cars and pedestrians using YOLOv5

### PR DESCRIPTION
**Pull request by fcakyon**
fixes https://github.com/tryolabs/norfair/issues/50

### CODE REVIEW

**Comment by fcakyon** [comment]
@joaqo @dekked I think its ready. Would you mind reviewing it?

**Comment by joaqo** [comment]
Hi @fcakyon, this is absolutely fantastic! Please give us a bit of time to look at it so we can start the process of getting it merged, it looks really good so the process should be quite fast though. Thank you!

File: README.md [hunk 1]
```diff
@@ -101,9 +101,10 @@ You can find the documentation for Norfair's API [here](docs/README.md).
 We provide several examples of how Norfair can be used to add tracking capabilities to several different detectors.
 
 1. [Simple tracking of cars](demos/detectron2) using [Detectron2](https://github.com/facebookresearch/detectron2).
-2. [Faster tracking of cars](demos/yolov4) using [YOLOv4](https://github.com/Tianxiaomo/pytorch-YOLOv4/tree/master). Try it on any youtube video on this [Google Colab notebook](https://colab.research.google.com/github/tryolabs/norfair/blob/master/demos/yolov4/yolov4_demo.ipynb).
-3. Inserting Norfair into an existing project: [Simple tracking of pedestrians](demos/alphapose) using [AlphaPose](https://github.com/MVIG-SJTU/AlphaPose).
-4. [Speed up pose estimation by extrapolating detections](demos/openpose) using [OpenPose](https://github.com/CMU-Perceptual-Computing-Lab/openpose).
+2. [Faster tracking of cars/pedestrians and other 78 classes](demos/yolov5) using [YOLOv5](https://github.com/fcakyon/yolov5-pip). Try it on any youtube video on this [Google Colab notebook](https://colab.research.google.com/github/tryolabs/norfair/blob/master/demos/yolov5/yolov5_demo.ipynb).
```
Comment (dekked) [inline README.md:104]: Can we capitalize YouTube?

(yes I know it's broken from before, but let's take the chance :P)

File: README.md [hunk 2]
```diff
@@ -113,12 +114,13 @@ Norfair's contribution to Python's object tracker library repertoire is its abil
...
-- [**SORT**](https://github.com/abewley/sort) and [**Deep SORT**](https://github.com/nwojke/deep_sort) are similar to this repo in that they use Kalman filters (and a deep embedding for Deep SORT), but they are hardcoded to a fixed distance function and to tracking boxes. Norfair also adds some filtering when matching tracked objects with detections, and changes the Hungarian Algorithm for its own distance minimizer. Both these repos are also released under the GPL license, which might be an issue for some individuals or companies because the source code of derivative works needs to be published.
+-   [**OpenCV**](https://opencv.org) includes several tracking solutions like [KCF Tracker](https://docs.opencv.org/3.4/d2/dff/classcv_1_1TrackerKCF.html) and [MedianFlow Tracker](https://docs.opencv.org/3.4/d7/d86/classcv_1_1TrackerMedianFlow.html) which are run by making the user select a part of the frame to track, and then letting the tracker follow that area. They tend not to be run on top of a detector and are not very robust.
```
Comment (dekked) [inline README.md:117]: Undo these modifications as well.

File: demos/yolov5/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+<h1 align="center"> YOLOv5 Example </h1>
```
Comment (dekked) [inline demos/yolov5/README.md:1]: Can we switch to standard markdown as the other demos?

Let's not use centered HTML titles for consistency.

File: demos/yolov5/yolov5_demo.ipynb [hunk 4]
```diff
@@ -0,0 +1,217 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "view-in-github"
+   },
+   "source": [
+    "<a href=\"https://colab.research.google.com/github/tryolabs/norfair/blob/master/demos/yolov4/yolov4_demo.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```
Comment (dekked) [inline demos/yolov5/yolov5_demo.ipynb:10]: Check the references to `yolov4` in the source of this file.

Looks like more than one link needs updating!

**Review by dekked — CHANGES_REQUESTED** [review]

**Comment by dekked** [comment]
Hi @fcakyon.

Thanks for your contribution, this looks great overall!

I have left a couple of comments. @joaqo feel free to leave more.

File: README.md [hunk 5]
```diff
@@ -8,13 +8,13 @@ Using Norfair, you can add tracking capabilities to any detector with just a few
 
 ## Features
 
-- Any detector expressing its detections as a series of `(x, y)` coordinates can be used with Norfair. This includes detectors performing object detection, pose estimation, and instance segmentation.
+-   Any detector expressing its detections as a series of `(x, y)` coordinates can be used with Norfair. This includes detectors performing object detection, pose estimation, and instance segmentation.
```
[the remark this answers is not in the document]
Comment (fcakyon) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (fcakyon) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (fcakyon) [reply README.md]: done

File: demos/yolov5/yolov5_demo.ipynb (hunk 4 above)
Comment (fcakyon) [reply demos/yolov5/yolov5_demo.ipynb]: done

**Comment by fcakyon** [comment]
@dekked all done :)

File: demos/yolov5/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+## Explanation
+
+This example tracks objects using a single point per detection: the centroid of the bounding boxes around objects returned by YOLOv5.
+
+## Car tracking demo:
+
+<img src="https://github.com/fcakyon/public-files/raw/main/norfair/yolov5_cars.gif" width="800" >
+
+## Pedestrian tracking demo: </h2>
```
Comment (dekked) [inline demos/yolov5/README.md:20]: There are some leftover `</h2>` here :)

File: README.md [hunk 7]
```diff
@@ -8,13 +8,13 @@ Using Norfair, you can add tracking capabilities to any detector with just a few
 
 ## Features
 
-- Any detector expressing its detections as a series of `(x, y)` coordinates can be used with Norfair. This includes detectors performing object detection, pose estimation, and instance segmentation.
+-  Any detector expressing its detections as a series of `(x, y)` coordinates can be used with Norfair. This includes detectors performing object detection, pose estimation, and instance segmentation.
```
Comment (dekked) [inline README.md:11]: These extra spaces somehow slipped by, can you make sure they are undone?

File: README.md [hunk 8]
```diff
@@ -101,9 +101,10 @@ You can find the documentation for Norfair's API [here](docs/README.md).
 We provide several examples of how Norfair can be used to add tracking capabilities to several different detectors.
 
 1. [Simple tracking of cars](demos/detectron2) using [Detectron2](https://github.com/facebookresearch/detectron2).
-2. [Faster tracking of cars](demos/yolov4) using [YOLOv4](https://github.com/Tianxiaomo/pytorch-YOLOv4/tree/master). Try it on any youtube video on this [Google Colab notebook](https://colab.research.google.com/github/tryolabs/norfair/blob/master/demos/yolov4/yolov4_demo.ipynb).
-3. Inserting Norfair into an existing project: [Simple tracking of pedestrians](demos/alphapose) using [AlphaPose](https://github.com/MVIG-SJTU/AlphaPose).
-4. [Speed up pose estimation by extrapolating detections](demos/openpose) using [OpenPose](https://github.com/CMU-Perceptual-Computing-Lab/openpose).
+2. [Faster tracking of cars/pedestrians and other 78 classes](demos/yolov5) using [YOLOv5](https://github.com/fcakyon/yolov5-pip). Try it on any Youtube video on this [Google Colab notebook](https://colab.research.google.com/github/tryolabs/norfair/blob/master/demos/yolov5/yolov5_demo.ipynb).
```
Comment (dekked) [inline README.md:104]: Actually it should be "YouTube" :)

File: README.md [hunk 9]
```diff
@@ -113,12 +114,13 @@ Norfair's contribution to Python's object tracker library repertoire is its abil
...
-- [**SORT**](https://github.com/abewley/sort) and [**Deep SORT**](https://github.com/nwojke/deep_sort) are similar to this repo in that they use Kalman filters (and a deep embedding for Deep SORT), but they are hardcoded to a fixed distance function and to tracking boxes. Norfair also adds some filtering when matching tracked objects with detections, and changes the Hungarian Algorithm for its own distance minimizer. Both these repos are also released under the GPL license, which might be an issue for some individuals or companies because the source code of derivative works needs to be published.
+-  [**OpenCV**](https://opencv.org) includes several tracking solutions like [KCF Tracker](https://docs.opencv.org/3.4/d2/dff/classcv_1_1TrackerKCF.html) and [MedianFlow Tracker](https://docs.opencv.org/3.4/d7/d86/classcv_1_1TrackerMedianFlow.html) which are run by making the user select a part of the frame to track, and then letting the tracker follow that area. They tend not to be run on top of a detector and are not very robust.
```
Comment (dekked) [inline README.md:117]: Again extra spaces.

File: demos/yolov5/README.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
+# YOLOv5 Example
+
+Simplest possible example of tracking. Based on <a href="https://github.com/fcakyon/yolov5-pip">YOLOv5-pip</a>.
```
Comment (dekked) [inline demos/yolov5/README.md:3]: Can we make this a markdown link instead instead of HTML?

File: demos/yolov5/README.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+## Instructions 
+
+1. Install Norfair with `pip install norfair[video]`.
+2. Install YOLOv5 with `pip install yolov5`.
+3. Copy `yolov5demo.py` into your local and run `python yolov5demo.py <video file>`.
+4. Bonus: Use additional arguments `--detector_path`, `--img_size`, `--iou_thres`,`--conf_thres`, `--classes`, `--track_points` as you wish.
+
+## Explanation
+
+This example tracks objects using a single point per detection: the centroid of the bounding boxes around objects returned by YOLOv5.
```
Comment (dekked) [inline demos/yolov5/README.md:14]: This is now only partially correct since you added the `--track_points` option ;)

**Review by dekked — CHANGES_REQUESTED** [review]

**Comment by fcakyon** [comment]
@dekked all done :)

**Comment by fcakyon** [comment]
@dekked i guess it would be better to draw boxes when `track_points` arg is set as `bbox`. do you have any demo with `draw_boxes`? what is the input `frame` for function `draw_boxes`?

**Comment by fcakyon** [comment]
@dekked all updated :)

**Review by DonBraulio — APPROVED** [review]
Looks good to me.
Thanks @fcakyon for this awesome contribution!!! 👏

**Review by dekked — APPROVED** [review]

**Comment by joaqo** [comment]
Thanks a lot for this PR, its great!

**Pull request merged**